### PR TITLE
Add filter and csv option to list command; allow filtering on attributes

### PR DESCRIPTION
**Pull request by dawzee**

### CODE REVIEW

File: req.go [hunk 1]
```diff
@@ -783,10 +783,101 @@ type ReqFilter struct {
 	AttributeRegexp    map[string]*regexp.Regexp
 }
 
+// createFilterFromCmdLine reads the filter regular expressions from the command line arguments and
+// compiles them into a filter structure ready to use
+// @llr REQ-TRAQ-SWL-19, REQ-TRAQ-SWL-73
+func createFilterFromCmdLine(idFilter, titleFilter, bodyFilter string, attributeFilter []string) (ReqFilter, error) {
```
Comment (Javier-varez) [inline req.go:789]: `FromCmdLine` doesn't seem very accurate anymore. Maybe just remove it or use `FromParts`.

File: req.go [hunk 2]
```diff
@@ -783,10 +783,101 @@ type ReqFilter struct {
...
+		for _, f := range attributeFilter {
+			if strings.Contains(f, "=") {
+				parts := strings.Split(f, "=")
+				r, err := regexp.Compile(parts[1])
+				if err != nil {
+					return filter, err
+				}
+				filter.AttributeRegexp[strings.ToUpper(parts[0])] = r
+			} else {
+				filter.AnyAttributeRegexp, err = regexp.Compile(f)
```
Comment (Javier-varez) [inline req.go:821]: we should probably trigger an error if the user specifies more than one `any` attribute.

**Review by Javier-varez — APPROVED** [review]
Thanks, mostly minor comments.

File: config/config.go [hunk 3]
```diff
@@ -192,7 +192,7 @@ func (rs ReqSpec) ToString() string {
 func (config *Config) FindCertdoc(path string) (repos.RepoName, *Document) {
 	for repoName := range config.Repos {
 		for docIdx := range config.Repos[repoName].Documents {
-			if config.Repos[repoName].Documents[docIdx].Path == path {
+			if strings.Contains(config.Repos[repoName].Documents[docIdx].Path, path) {
```
[the remark this answers is not in the document]
Comment (dawzee) [reply config/config.go:195]: yeah fair enough, I was trying to fix the fact that you couldn't run "list" or "nextid" unless you were in the base directory, but this is probably too far the other way. I've gone with a strict comparison of the basename as a compromise.

**Pull request merged**